specs/interop/liquidity_migration.md

```diff
@@ -0,0 +1,389 @@
+
+1. Valid legacy address:
+   Verify the `OptimismMintableERC20Factories` deployment lists. This approach requires updating
+   the `OptimismMintableERC20Factory` to store
+   deployments as a mapping from `OptimismMintableERC20`
+   to its `REMOTE_TOKEN`.  
+    The mapping will be backwards compatible, as detailed in the [Appendix](#backwards-compatibility).
+2. Valid `SuperchainERC20` address:
+   checks the `SuperchainERC20Factory` mapping
+   that stores the corresponding remote token address
+   for each `SuperchainERC20` address.
+3. Both tokens correspond to the same remote token address:
+   it will compare the remote token addresses stored in the deployment
+   mappings and revert with `InvalidPair` if they do not match.
+4. Both tokens have the same amount of decimals.
+
+You can read more details about access control in the [Appendix](#decimal-management).
+
+A reference implementation would look like follows
+
+```solidity
+IOptimismMintableERC20Factory public factory = IOptimismMintableERC20Factory(Predeploys.OptimismMintableERC20Factory);
+ISuperchainERC20Factory public superFactory = ISuperchainERC20Factory(Predeploys.SuperchainERC20Factory);
+
+function checkPair(address _legacyAddr, address _superAddr) internal view returns (bool) {
+  // Check validity of the legacy token
+  address _legacyRemoteToken = factory.deployments(_legacyAddr);
+  require(_legacyRemoteToken != address(0), "Invalid Legacy address");
+
+  // Check validity of the superchainERC20 token
+  address _superRemoteToken = superFactory.deployments(_superAddr);
+  require(_superRemoteToken != address(0), "Invalid SuperchainERC20 address");
+
+  // Check both tokens correspond to the same remote token
+  require(_legacyRemoteToken == _superRemoteToken, "Invalid token pair");
+
+  // Check both tokens share decimals
+  require(IERC20Metadata(_legacyAddr).decimals() == IERC20Metadata(_superAddr).decimals(), "Decimals do not match")
+
+  return true;
+}
+```
+
+Note: Code can be simplified by using || in
+the requires or conditionals.
+The tradeoff would be losing error precision.
+
+### OptimismMintableERC20Factory
+
+#### `deployments`
+
+The `OptimismMintableERC20Factory` SHOULD include a new mapping that stores the remote
+token address for each deployed `OptimismMintableERC20`.
+
+```solidity
+mapping(address => address) public deployments;
+```
+
+The mapping will be updated on new updates and only once
+manually via storage manipulation to include the list of allowed tokens
+(see the [Appendix](#backwards-compatibility)).
+
+#### `createOptimismMintableERC20`
+
+The function should be modified to update the `deployments` mapping.
+
+```solidity
+function createOptimismMintableERC20(address _remoteToken, string memory _name, string memory _symbol) public returns (address) {
+  require(_remoteToken != address(0), "OptimismMintableERC20Factory: must provide remote token address");
+  bytes32 salt = keccak256(abi.encode(_remoteToken, _name, _symbol));
```

Comment:
I think that we can leave out most of this pseudocode and just define the salt and the new `deployments` mapping

Reply:
Done